crates/subspace-node/src/commands/run.rs

```diff
@@ -113,7 +114,20 @@ pub async fn run(run_options: RunOptions) -> Result<(), Error> {
     info!("🏷  Node name: {}", subspace_configuration.network.node_name);
     info!("💾 Node path: {}", base_path.display());
 
-    let consensus_state_pruning = subspace_configuration
+    if maybe_domain_configuration.is_some()
+        && matches!(
+            subspace_configuration.state_pruning,
+            Some(PruningMode::Constrained(_))
+        )
```

Comment:
Should blocks also be archival for operators? Operator documentation didn't have it and default was constrained for a very long time. Note that node will finalize blocks at a significant depth, so it may not be a problem anyway.

Reply:
>Should blocks also be archival for operators?

Yeah, some FP requires getting the bundle from the consensus block body, I know the consensus block only prunes after finalization but since the challenge period is measured by domain block it is still possible for an operator fail to generate FP after go offline for a long time.